odl/discr/lp_discr.py

```diff
@@ -446,8 +528,16 @@ def __setitem__(self, indices, values):
             shape is allowed as ``values``.
         """
         if values in self.space:
-            self.ntuple.__setitem__(indices, values.ntuple)
+            # For RawDiscretizationVector of the same type, use ntuple directly
+            return self.ntuple.__setitem__(indices, values.ntuple)
+        elif (isinstance(self.space.dspace, Ntuples) and
+              equiv_views(self.asarray(), values) and
```

Comment:
> Also, if we know that self.space.dspace is a Ntuples, we should use self.data instead since this is both faster and more clear.

I'll change that. Otherwise, ok with a TODO?